src/types/dataModel/eventVolunteer.ts

```diff
@@ -0,0 +1,41 @@
+import { z } from 'zod';
+import zVolunteer, { zRole, zVolunteerResponse } from './volunteer';
+import zOrganization from './organization';
+import zBase, { zObjectId } from './base';
+
+const zEventVolunteer = z.object({
+  role: zRole,
+  organization: zOrganization.optional(),
+  volunteer: zVolunteer,
+  event: zObjectId, //TODO: Replace with zEvent
+});
+
+const zEventVolunteerEntity = zEventVolunteer.extend({
+  ...zBase.shape,
+  role: zObjectId,
```

Comment:
you don't need to make this an object id, its fine to be a `zRole` in the entity